url8551e9ebc1_mishika05_Salesforce_DataGen_tgz-pJ8-org_json_TestJ8/src/RecordGen.java

```diff
@@ -0,0 +1,68 @@
+    
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.io.File;
+import java.io.FileReader;
+import java.io.LineNumberReader;
+import java.text.NumberFormat;
+import org.apache.log4j.Logger;
+
+/**
+ * The RecordGen class uses Record.java, SalesforceClient.java
+ * and WorkerThread.java to push numBugs # of Records (Bugs) to your Salesforce app.
+ * @author  Mishika Vora
+ */
+public class RecordGen {
+	private static Logger recIdsLogger = Logger.getLogger("recIdsLogger");
+	private static Logger logger = Logger.getLogger(RecordGen.class);
+	public static long startTime;
+    public static long endTime;
+
+
+    public static void main(String[] args) throws Exception {
+    	String propFile;
+    	if (args.length > 0) {
+    	    propFile = args[0];
+    	}
+    	else 
+    		propFile = Constants.defaultPropFile;    
+   	 	 SalesForceClient client = new SalesForceClient(propFile);
+    	 ExecutorService executor = Executors.newFixedThreadPool(client.prop.numThreads);
+    	 Record rec = new Bug(client);
+    	 startTime = System.nanoTime();
+    	 for (int i = 0; i < client.prop.numRecs ; i++) {
+    		 Runnable worker = new WorkerThread(rec,"" + i);
+    		 executor.execute(worker);
+    	 }
+    	 executor.shutdown();
+    	 while (!executor.isTerminated()) {
+    	    
+    	 }
+      	 endTime = System.nanoTime();
+    	 logger.info("Finished all Threads");
+    	 LineNumberReader  lnr = new LineNumberReader(new FileReader(new File("bugIds.log")));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *lnr*. The resource is referenced at lines 44, 45, 46. The resource is closed at line 59. There are other execution paths that do not contain closure statements, for example, when LineNumberReader.skip throws an exception. Ensure *lnr* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).